### DISCUSSION TITLE
《AI：梦境档案》简体中文补丁

### DISCUSSION THREAD

**Discussion by giscus-xzonn** [Xzonn/xzonn.github.io#8] [General]
# 《AI：梦境档案》简体中文补丁

《AI：梦境档案》简体中文补丁 - 来自Xzonn的小站

https://xzonn.top/posts/AI-The-Somnium-Files-Chs-Patch.html

**Comment by Xzonn** [comment]
> 评论导出自来必力

*****Em Ewango 评论于 2022-10-17 05:29:33 (UTC+8)*****

你好﹐我正在尝试制作类似于文中的档案﹐但当中有些不明白的地方。  
在开始之前我已经看过全部内容。我先尝试不同步骤﹐打算最后才真的制作内容。而我正在制作的是This is the police 2游戏。当中文件结构似乎很相似﹐都是以font文件储存在StreamingAssets\AssetBundles之中。  
当中最大的问题有三个：  
一．生成新的的MonoBehaviour和Texture2D  
是否必须让新旧两个Texture2D文件的文字位置相对应？因为当中似乎涉及使用python使用﹐我不熟悉这个步骤。而且我希望由简体转为繁体﹐即将会有更多字码﹐对应不了全部位置。  
相反﹐可否直接使用自己不按旧有资料所生成的MonoBehaviour和Texture2D档案？还是我对过程有所误会？

二． Import Dump功能  
Import Dump可否直接Import .png档？因为export dump 的Texture2D是.txt 格式﹐而要用Plugins功能获得 .png档。我也不清楚Import Dump这功能的运作原理

三． resources.assets部分文字  
这部分的文字是否必须执行同样操作以达到字码转换的目的？

最后请问是否方便使用不同的通讯方式？还是可以在这里讨论？谢谢！

*****Xzonn 回复于 2022-10-24 23:00:49 (UTC+8)*****

感谢提问，由于评论系统最近电子邮件通知有故障，我可能无法及时回复。  
关于这几个问题，因为我本人之前学过一些数据结构方面的知识，所以文章里面写得比较简略。而你提到的这个游戏我本人也没玩过，我所说的情况也不一定适用于这个游戏。我简单解答一下：  
一、新字体的生成是用 Unity 自带的 TextMesh Pro 插件实现的，不同游戏使用的插件版本不一样，需要选择和游戏中使用的插件版本一致才能进行后续步骤。生成的新字体与旧字体不必文字位置一一对应，因为后续有替换操作，对应本文“替换掉之前导出的MonoBehaviour的json文件的对应值”一句。替换操作工作量比较大，我是用 Python 完成的。  
二、Import Dump 实质上是导入了图片文件的基本信息，而不包含图片的像素点数据，这部分数据一般需要通过替换 resS 文件实现。不过我最近也发现有游戏没有放到 resS 文件而是直接和 Texture2D 的基本信息放一起，这种情况就可以通过修改 json 文件实现。  
三、我在前期操作的时候并没有修改文本，这也正是我所说的“替换字形”的意思，也就是说数据中保存的文本仍然是繁体中文，只是我让它显示成了简体中文。对于你的情况可能是正好反过来。当然，也可以把修改后的文本导入游戏，这样的话就只需要把旧字体的全部信息修改为新字体的信息即可，无需做字符码的转换。  
如果需要即时聊天的话，可以在 Bilibili 私信我（<https://space.bilibili.com/16114399>），或者通过 Twitter（<https://twitter.com/Xzonn2000>）。
